modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/internal/UncancelablePromise.kt

```diff
@@ -6,7 +6,7 @@ import arrow.effects.Promise
 import arrow.effects.typeclasses.Async
 import java.util.concurrent.atomic.AtomicReference
 
-internal class UncancelablePromise<F, A>(AS: Async<F>) : Promise<F, A>, Async<F> by AS {
+internal class UncancelablePromise<F, A>(private val AS: Async<F>) : Promise<F, A>, Async<F> by AS {
```

Comment:
This shouldn't be necessary, it should be an import alias instead, right?

Reply:
There are multiple inheritance problems from `: Async`, if you don't explicitly override those methods it'll give a warning.